src/libraries/System.Data.Odbc/src/Common/System/Data/Common/AdapterUtil.Odbc.cs

```diff
@@ -663,7 +663,7 @@ internal static IntPtr IntPtrOffset(IntPtr pbase, int offset)
                 return (IntPtr)checked(pbase.ToInt32() + offset);
             }
             Debug.Assert(8 == ADP.PtrSize, "8 != IntPtr.Size"); // MDAC 73747
-            return (IntPtr)checked(pbase.ToInt64() + offset);
+            return checked((IntPtr)(pbase.ToInt64() + offset));
```

Comment:
This is simpler and matches the intent:
```suggestion
            return checked(pbase + offset);
```

Reply:
The `checked` arithmetic is a bug. It will fail in processes with large address space. This whole function should be just 

```
internal static IntPtr IntPtrOffset(IntPtr pbase, int offset)
{
    return pbase + offset;
}
```

It would be even better to change this to use `void*`, but that would be more invasive change.

Reply:
Actually, this project targets both .NET Framework and .NET Core. Does it mean that the code has to work for both with IntPtr-arithmetic-is-checked and IntPtr-arithmetic-is-unchecked?

If it is the case, my suggestion won't work. I think it needs to be something like:
```
internal static IntPtr IntPtrOffset(IntPtr pbase, int offset)
{
    return (nint)pbase + offset;
}
```
I assume that casting to nint will suppress the checked IntPtr arithmetic, correct?

Reply:
Correct